x-pack/legacy/plugins/ml/public/application/jobs/new_job/pages/components/job_details_step/components/additional_section/components/calendars/calendars_selection.tsx

```diff
@@ -5,49 +5,52 @@
  */
 
 import React, { FC, useState, useContext, useEffect } from 'react';
-import { EuiComboBox, EuiComboBoxOptionProps } from '@elastic/eui';
+import { EuiComboBox, EuiComboBoxOptionProps, EuiComboBoxProps } from '@elastic/eui';
 import { JobCreatorContext } from '../../../../../job_creator_context';
 import { Description } from './description';
 import { ml } from '../../../../../../../../../services/ml_api_service';
+import { Calendar } from '../../../../../../../../../../../common/types/calendars';
 
 export const CalendarsSelection: FC = () => {
   const { jobCreator, jobCreatorUpdate } = useContext(JobCreatorContext);
-  const [selectedCalendars, setSelectedCalendars] = useState(jobCreator.calendars);
-  const [selectedOptions, setSelectedOptions] = useState<EuiComboBoxOptionProps[]>([]);
-  const [options, setOptions] = useState<EuiComboBoxOptionProps[]>([]);
+  const [selectedCalendars, setSelectedCalendars] = useState<Calendar[]>(jobCreator.calendars);
+  const [selectedOptions, setSelectedOptions] = useState<Array<EuiComboBoxOptionProps<Calendar>>>(
+    []
+  );
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionProps<Calendar>>>([]);
   const [isLoading, setIsLoading] = useState(false);
 
   async function loadCalendars() {
     setIsLoading(true);
     const calendars = await ml.calendars();
-    setOptions(calendars.map(c => ({ label: c.calendar_id })));
-    setSelectedOptions(selectedCalendars.map(c => ({ label: c })));
+    setOptions(calendars.map(c => ({ label: c.calendar_id, value: c })));
+    setSelectedOptions(selectedCalendars.map(c => ({ label: c.calendar_id, value: c })));
     setIsLoading(false);
   }
 
   useEffect(() => {
     loadCalendars();
   }, []);
 
-  function onChange(optionsIn: EuiComboBoxOptionProps[]) {
-    setSelectedOptions(optionsIn);
-    setSelectedCalendars(optionsIn.map(o => o.label));
-  }
-
   useEffect(() => {
     jobCreator.calendars = selectedCalendars;
     jobCreatorUpdate();
   }, [selectedCalendars.join()]);
 
+  const comboBoxProps: EuiComboBoxProps<Calendar> = {
+    async: true,
+    options,
+    selectedOptions,
+    isLoading,
+    onChange: optionsIn => {
+      setSelectedOptions(optionsIn);
+      setSelectedCalendars(optionsIn.map(o => o.value as Calendar));
+    },
+  };
+
   return (
     <Description>
-      <EuiComboBox
-        async
-        options={options}
-        selectedOptions={selectedOptions}
-        isLoading={isLoading}
-        onChange={onChange}
-      />
+      <EuiComboBox {...comboBoxProps} />
```

Comment:
I think it would be good to have a 'Manage calendars' link to the calendars settings page next to the calendars control, which would open the settings page in a new window. There would also need to be a refresh button ('EuiButtonIcon` ?) which allows the user to see the newly created calendar in the `EuiComboBox`.

Reply:
Check https://github.com/elastic/kibana/pull/51281/commits/a5331aa68a9cda274a531c867f0e9d2814617c7d and https://github.com/elastic/kibana/pull/51281/commits/7e5b22e2cdc8461ebf942febd752f9233e07d761